examples/quantize/quantize.cpp

```diff
@@ -159,18 +162,34 @@ static void load_imatrix(const std::string & imatrix_file, std::unordered_map<st
             printf("%s: loaded data (size = %6d, ncall = %6d) for '%s'\n", __func__, int(e.size()), ncall, name.c_str());
         }
     }
-    printf("%s: loaded %d importance matrix entries from %s\n", __func__, int(imatrix_data.size()), imatrix_file.c_str());
+
+    // latest imatrix version contains the dataset filename at the end of the file
+    int m_last_call = 0;
+    if (in.peek() != EOF) {
+        in.read((char *)&m_last_call, sizeof(m_last_call));
+        int dataset_len;
+        in.read((char *)&dataset_len, sizeof(dataset_len));
+        std::vector<char> dataset_as_vec(dataset_len+1);
+        in.read((char *)dataset_as_vec.data(), dataset_len);
+        dataset_as_vec[dataset_len] = 0;
+        imatrix_dataset = std::string{dataset_as_vec.data()};
```

Comment:
There is no need to add a NUL to the `std::vector`, `std::string` can be initialized from a pair of iterators.
```suggestion
        std::vector<char> dataset_as_vec(dataset_len);
        in.read(dataset_as_vec.data(), dataset_len);
        imatrix_dataset.assign(dataset_as_vec.begin(), dataset_as_vec.end());
```